gremlin-go/driver/connection.go

```diff
@@ -66,17 +67,16 @@ func (connection *connection) write(request *request) (ResultSet, error) {
 	return connection.results[requestID], connection.protocol.write(request)
```

Comment:
ResultSet is an interface and the implementation in channelResultSet, this is because it uses a channel under the hood. The reason for the naming is because it makes a channel based result set, I like it this way because if we wanted to we could change to a slice based ResultSet and it would be newSliceResultSet.